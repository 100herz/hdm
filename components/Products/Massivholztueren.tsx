import styled from "styled-components"

const Grid = styled.div`
  display: grid;
  grid-template-areas:
    "grid-item-1"
    "grid-item-2"
    "grid-item-3";
  grid-template-columns: 1fr;

  @media (min-width: 768px) {
    grid-template-areas:
      "grid-item-1 grid-item-2"
      "grid-item-3 grid-item-2";
    grid-template-columns: 1fr 1fr;
    column-gap: 8rem;
  }

  & > :nth-child(1) {
    grid-area: grid-item-1;
  }

  & > :nth-child(2) {
    grid-area: grid-item-2;
  }

  & > :nth-child(3) {
    grid-area: grid-item-3;
  }
`

type Props = {
  className?: string
}

const Massivholztueren: React.FC<Props> = ({ className = "" }: Props) => {
  return (
    <section className={`overflow-x-hidden relative lg:mt-24 ${className}`}>
      <Grid className="container pt-20 pb-20 bg-gray-100">
        <header className="lg:ml-32 lg:mt-8">
          <div>
            <h2 className="text-4xl font-bold lg:text-5xl">Massivholztüren</h2>
          </div>
        </header>

        <picture className="block mt-16 -mx-6 lg:ml-0 lg:-mr-32 lg:mt-0">
          <source srcSet="/images/hdm-massivholztueren.webp" type="image/webp" />
          <source srcSet="/images/hdm-massivholztueren.png" type="image/png" />
          <img
            src="/images/hdm-massivholztueren.png"
            alt="Massivholztüren - Sorglos die Schönheit massiven Holzes genießen"
          />
        </picture>

        <div className="lg:-mt-32 lg:ml-32">
          <p className="mt-12 lg:text-lg">
            Sorglos die Schönheit massiven Holzes genießen. Massivholz sorgt für behagliche Wärme und ist nie
            langweilig.
          </p>
          <p className="mt-8 lg:text-lg">
            Denn Holz ist pure Natur und bringt eine angenehme, freundliche Atmosphäre ins Haus, unterstrichen von
            ruhigen Licht- und Schattenspielen der offenen Lamellen- und Gittertüren.
          </p>

          <div className="mt-16 text-center lg:text-left lg:mt-24">
            <a
              href="/documents/hdm-cerastar.pdf"
              target="_blank"
              title="Den Katalog Cerastar ansehen und herunterladen"
              className="inline-block px-6 py-3 text-sm text-white rounded-full bg-hdm"
            >
              <svg
                className="inline-block mr-3"
                xmlns="http://www.w3.org/2000/svg"
                width="17.588"
                height="17.47"
                viewBox="0 0 17.588 17.47"
              >
                <path
                  d="M112.359,182.379a1.787,1.787,0,0,1-1.21-.524,2.06,2.06,0,0,1-.565-1.936,1.589,1.589,0,0,1,.686-1.008,19,19,0,0,1,4.113-2.3,44.427,44.427,0,0,0,2.419-5.041,11.514,11.514,0,0,1-1.29-3.75,3.57,3.57,0,0,1,.121-1.613c.4-1.29,1.29-1.331,1.653-1.29.081,0,1.492.2,1.734,2.218a10.9,10.9,0,0,1-.927,4.4,14,14,0,0,0,3.145,3.71,18.613,18.613,0,0,1,4.557.524,1.715,1.715,0,0,1,.927.565,1.5,1.5,0,0,1,.282,1.573h0a1.675,1.675,0,0,1-.323.565,1.558,1.558,0,0,1-1.976.444,21.456,21.456,0,0,1-3.871-2.54,16.963,16.963,0,0,0-5.565,1.169c-.686,1.25-1.532,2.621-2.5,4.153a1.7,1.7,0,0,1-.766.645h0C112.722,182.339,112.521,182.379,112.359,182.379Zm1.976-4.073a15.061,15.061,0,0,0-2.379,1.492c-.121.121-.242.242-.242.363a.956.956,0,0,0,.323.887.574.574,0,0,0,.484.121h0a.331.331,0,0,0,.2-.2C113.287,180.04,113.851,179.153,114.335,178.306Zm9.4-1.936a22.573,22.573,0,0,0,2.419,1.452.385.385,0,0,0,.484-.121,1.115,1.115,0,0,0,.161-.282c.081-.2.081-.282-.04-.4a.78.78,0,0,0-.363-.2A15.283,15.283,0,0,0,123.731,176.371Zm-5.242-3.589c-.4.927-.887,1.976-1.532,3.186a18.4,18.4,0,0,1,3.549-.726A20.147,20.147,0,0,1,118.489,172.782Zm-.4-6.775c-.161,0-.282.161-.4.524a2.408,2.408,0,0,0-.081,1.089,10.078,10.078,0,0,0,.686,2.339,8.044,8.044,0,0,0,.524-2.742c-.121-1.089-.686-1.21-.726-1.21Z"
                  transform="translate(-110.524 -164.909)"
                  fill="#fff"
                />
              </svg>
              Massivholztüren Katalog downloaden
            </a>
          </div>
        </div>
      </Grid>
    </section>
  )
}

export default Massivholztueren
