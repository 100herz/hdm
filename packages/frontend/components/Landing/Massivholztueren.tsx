import dynamic from 'next/dynamic'
// import Image from 'next/image'
import styled from '@emotion/styled'

export default function Massivholztueren({ className = '' }: React.HTMLAttributes<HTMLElement>) {
  const Pdf = dynamic(() => import('@/components/Utils/Pdf'), { ssr: false })

  return (
    <section id="massivholztueren" className={`overflow-x-hidden relative lg:mt-24 ${className}`}>
      <Grid className="container pt-20 pb-20 bg-gray-100">
        <header className="lg:ml-32 lg:mt-8 first">
          <h2 className="text-4xl font-bold lg:text-5xl">Massivholztüren</h2>
          <p className="mt-4 text-xl leading-snug lg:text-2xl">
            Das Lamellen- und <br /> Gittertüren Programm
          </p>
        </header>

        {/* TODO: `Image` is currently not available in SSG */}
        {/* <div className="mt-16 -mx-6 lg:ml-0 lg:-mr-32 lg:mt-0 second">
          <Image
            src="/images/hdm-massivholztueren.png"
            alt="Massivholztüren - Sorglos die Schönheit massiven Holzes genießen"
            width={768}
            height={779}
          />
        </div> */}
        <picture className="block mt-16 -mx-6 lg:ml-0 lg:-mr-32 lg:mt-0 second">
          <source srcSet="/images/hdm-massivholztueren.webp" type="image/webp" />
          <source srcSet="/images/hdm-massivholztueren.png" type="image/png" />
          <img
            src="/images/hdm-massivholztueren.png"
            alt="Massivholztüren - Sorglos die Schönheit massiven Holzes genießen"
            loading="lazy"
          />
        </picture>

        <div className="lg:-mt-16 lg:ml-32 third">
          <p className="mt-12 lg:text-lg">
            Sorglos die Schönheit massiven Holzes genießen. Denn Holz ist pure Natur und bringt eine angenehme,
            freundliche Atmosphäre ins Haus, unterstrichen von ruhigen Licht- und Schattenspielen der offenen Lamellen-
            und Gittertüren.
          </p>

          <p className="mt-8 lg:text-lg">Massivholz sorgt für behagliche Wärme und ist nie langweilig.</p>

          <div className="mt-16 text-center lg:text-left lg:mt-24">
            <Pdf title="Massivholztüren Katalog" filename="/documents/hdm-massivholztueren.pdf" />
          </div>
        </div>
      </Grid>
    </section>
  )
}

const Grid = styled.div`
  display: grid;
  grid-template-areas:
    'grid-item-1'
    'grid-item-2'
    'grid-item-3';
  grid-template-columns: 1fr;

  @media (min-width: 1024px) {
    grid-template-areas:
      'grid-item-1 grid-item-2'
      'grid-item-3 grid-item-2';
    grid-template-columns: 1fr 1fr;
    column-gap: 8rem;
  }

  & > .first {
    grid-area: grid-item-1;
  }

  & > .second {
    grid-area: grid-item-2;
  }

  & > .third {
    grid-area: grid-item-3;
  }
`
