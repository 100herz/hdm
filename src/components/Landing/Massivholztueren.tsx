import dynamic from "next/dynamic"
import styled from "styled-components"

const Grid = styled.div`
  display: grid;
  grid-template-areas:
    "grid-item-1"
    "grid-item-2"
    "grid-item-3";
  grid-template-columns: 1fr;

  @media (min-width: 1024px) {
    grid-template-areas:
      "grid-item-1 grid-item-2"
      "grid-item-3 grid-item-2";
    grid-template-columns: 1fr 1fr;
    column-gap: 8rem;
  }

  & > :nth-child(1) {
    grid-area: grid-item-1;
  }

  & > :nth-child(2) {
    grid-area: grid-item-2;
  }

  & > :nth-child(3) {
    grid-area: grid-item-3;
  }
`

type Props = {
  className?: string
}

const Massivholztueren: React.FC<Props> = ({ className = "" }: Props) => {
  const Pdf = dynamic(() => import("@/components/Utils/Pdf"), { ssr: false })

  return (
    <section id="massivholztueren" className={`overflow-x-hidden relative lg:mt-24 ${className}`}>
      <Grid className="container pt-20 pb-20 bg-gray-100">
        <header className="lg:ml-32 lg:mt-8">
          <h2 className="text-4xl font-bold lg:text-5xl">Massivholztüren</h2>
          <p className="mt-4 text-xl leading-snug lg:text-2xl">
            Das Lamellen- und <br /> Gittertüren Programm
          </p>
        </header>

        <picture className="block mt-16 -mx-6 lg:ml-0 lg:-mr-32 lg:mt-0">
          <source srcSet="/images/hdm-massivholztueren.webp" type="image/webp" />
          <source srcSet="/images/hdm-massivholztueren.png" type="image/png" />
          <img
            src="/images/hdm-massivholztueren.png"
            alt="Massivholztüren - Sorglos die Schönheit massiven Holzes genießen"
          />
        </picture>

        <div className="lg:-mt-16 lg:ml-32">
          <p className="mt-12 lg:text-lg">
            Sorglos die Schönheit massiven Holzes genießen. Denn Holz ist pure Natur und bringt eine angenehme,
            freundliche Atmosphäre ins Haus, unterstrichen von ruhigen Licht- und Schattenspielen der offenen Lamellen-
            und Gittertüren.
          </p>

          <p className="mt-8 lg:text-lg">Massivholz sorgt für behagliche Wärme und ist nie langweilig.</p>

          <div className="mt-16 text-center lg:text-left lg:mt-24">
            <Pdf title="Massivholztüren Katalog" filename="/documents/hdm-massivholztueren.pdf" />
          </div>
        </div>
      </Grid>
    </section>
  )
}

export default Massivholztueren
