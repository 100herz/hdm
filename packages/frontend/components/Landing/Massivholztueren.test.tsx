import { render } from '@/test/test-utils'

import Massivholztueren from './Massivholztueren'

describe('Massivholztueren component', () => {
  it('matches snapshot', () => {
    const { container, getByText } = render(<Massivholztueren />)
    expect(getByText('Massivholzt√ºren')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
