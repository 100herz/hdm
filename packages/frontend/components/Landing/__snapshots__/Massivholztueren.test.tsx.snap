// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Massivholztueren component matches snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-template-areas: 'grid-item-1' 'grid-item-2' 'grid-item-3';
  grid-template-columns: 1fr;
}

@media (min-width: 1024px) {
  .emotion-0 {
    grid-template-areas: 'grid-item-1 grid-item-2' 'grid-item-3 grid-item-2';
    grid-template-columns: 1fr 1fr;
    -webkit-column-gap: 8rem;
    column-gap: 8rem;
  }
}

.emotion-0>.first {
  grid-area: grid-item-1;
}

.emotion-0>.second {
  grid-area: grid-item-2;
}

.emotion-0>.third {
  grid-area: grid-item-3;
}

<section
  class="overflow-x-hidden relative lg:mt-24 "
  id="massivholztueren"
>
  <div
    class="container pt-20 pb-20 bg-gray-100 emotion-0 emotion-1"
  >
    <header
      class="lg:ml-32 lg:mt-8 first"
    >
      <h2
        class="text-4xl font-bold lg:text-5xl"
      >
        Massivholztüren
      </h2>
      <p
        class="mt-4 text-xl leading-snug lg:text-2xl"
      >
        Das Lamellen- und 
        <br />
         Gittertüren Programm
      </p>
    </header>
    <div
      class="mt-16 -mx-6 lg:ml-0 lg:-mr-32 lg:mt-0 second"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzY4IiBoZWlnaHQ9Ijc3OSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="Massivholztüren - Sorglos die Schönheit massiven Holzes genießen"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div
      class="lg:-mt-16 lg:ml-32 third"
    >
      <p
        class="mt-12 lg:text-lg"
      >
        Sorglos die Schönheit massiven Holzes genießen. Denn Holz ist pure Natur und bringt eine angenehme, freundliche Atmosphäre ins Haus, unterstrichen von ruhigen Licht- und Schattenspielen der offenen Lamellen- und Gittertüren.
      </p>
      <p
        class="mt-8 lg:text-lg"
      >
        Massivholz sorgt für behagliche Wärme und ist nie langweilig.
      </p>
      <div
        class="mt-16 text-center lg:text-left lg:mt-24"
      />
    </div>
  </div>
</section>
`;
